tests/page-objects/forms/generic-form.po.js

```diff
@@ -82,13 +81,13 @@ module.exports = {
   },
 
   selectFormNative: async (formId) => {
+    const addButton = element(by.css('.action-container .general-actions:not(.ng-hide) .fa-plus'));
     await helper.waitUntilReadyNative(addButton);
-    await helper.waitElementToBeClickable(addButton);
-    await addButton.click();
+
+    // select form
+    await helper.clickElement(addButton);
     const form = module.exports.formByHref(formId);
-    await helper.waitElementToPresentNative(form);
-    await form.click();
-    await helper.waitElementToPresentNative(element(by.css('#report-form')));
+    await helper.clickElement(form);
```

Comment:
You don't think it's necessary to wait until the form is loaded? 
Since you're removing 
```
await helper.waitElementToPresentNative(element(by.css('#report-form')));
```

Reply:
We could but we still need to await for an element in the injected form (xml) like a title or field.
Maybe we can send as parameter, a selector for the item inside this xml to await, so it's more clear usage, what do you think? 
`selectFormNative: async (formId, elementInForm) {.....}`

Reply:
Forms should always have a title, so we can go for 
```
await helper.waitElementToPresentNative(element(by.css('#report-form #form-title')));
```

Reply:
I think the above can be safely used here, as all forms should have titles, rather than in their individual tests.

Reply:
cool! done